sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/RateLimitedSampler.cs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using OpenTelemetry.Trace;
+using System.Diagnostics;
+using System.Linq;
+using System.Collections.Concurrent;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter.Internals
+{
+    internal class RateLimitedSampler : Sampler
+    {
+        private static readonly SamplingResult RecordOnlySamplingResult = new(SamplingDecision.RecordOnly);
+        private readonly RateLimitedSamplingPercentage _samplingPercentage;
+
+        public RateLimitedSampler(double targetSpansPerSecondLimit)
+        {
+            _samplingPercentage = new RateLimitedSamplingPercentage(targetSpansPerSecondLimit);
+        }
+
+        public override SamplingResult ShouldSample(in SamplingParameters samplingParameters)
+        {
+            // 1. Respect the sampling decision of the local parent span
+
+            // get the span context of the parent span
+            var parentContext = samplingParameters.ParentContext;
+
+            // grab the parent sampling rate if it exists
+            var parentAttributes = Activity.Current?.TagObjects;
+            object? parentSampleRate = parentAttributes?.FirstOrDefault(kv => kv.Key == "microsoft.sample_rate").Value;
```

Comment:
Consider using direct enumeration instead of LINQ for better performance:
```
foreach (var kvp in activity.TagObjects) {
    if (kvp.Key == "sample-rate") return kvp.Value;
}
```

Reply:
Above comment applies here too, fetch the value if parent context is valid and has sampled flag.